src/accelerate/utils/other.py

```diff
@@ -86,6 +89,21 @@ def extract_model_from_parallel(model, keep_fp32_wrapper: bool = True):
     while isinstance(model, options):
         model = model.module
 
+    if recursive:
+        # This is needed in cases such as using FSDPv2 on XLA
+        def _recursive_unwrap(module):
+            if hasattr(module, "module"):
```

Comment:
Could you add a comment on where `module.module` is coming from?

Reply:
Done